src/rust/sqs-executor/src/sqs_timeout_manager.rs

```diff
@@ -0,0 +1,186 @@
+use futures::future::{self, Either};
+use futures::pin_mut;
+use rusoto_sqs::{ChangeMessageVisibilityRequest, Sqs};
+
+use std::time::Duration;
+
+use tokio::stream::StreamExt;
+use tokio::sync::mpsc::{channel as mpsc_channel, Receiver as MpscReceiver, Sender as MpscSender};
```

Comment:
send-receive-future stuff - i'm just gonna say i'm not currently qualified to review this code

Reply:
update it's pretty readable except i don't know what the receiver is doing. but I'm assuming it's "ok i completed whatever processing I needed to do on this message receipt"